ta4j-core/src/test/java/org/ta4j/core/num/NumTest.java

```
@@ -227,5 +235,72 @@ public void testArithmetic(){
+    }
+
+    @Test
+    public void sqrtOfZero() {
+        assertNumEquals(0, numOf(0).sqrt(12));
+        assertNumEquals(0, numOf(0).sqrt());
+    }
+
+    @Test
+    public void sqrtLudicrousPrecision() {
+        BigDecimal numBD = BigDecimal.valueOf(Double.MAX_VALUE).multiply(BigDecimal.valueOf(Double.MAX_VALUE).add(BigDecimal.ONE));
+        Num sqrt = numOf(numBD).sqrt(100000);
+        if (numOf(0).getClass().equals(DoubleNum.class)) {
+            assertEquals("Infinity", sqrt.toString());
+        } else if (numOf(0).getClass().equals(BigDecimalNum.class)) {
+            assertNumEquals("1.7976931348623157000000000000000E+308", sqrt);
+            assertNumNotEquals("1.7976931348623157000000000000001E+308", sqrt);
+            assertNumEquals(Double.MAX_VALUE, sqrt);
+            assertNumEquals(numOf(Double.MAX_VALUE), sqrt);
+            BigDecimal sqrtBD = new BigDecimal(sqrt.toString());
+            assertNumEquals(numOf(numBD), numOf(sqrtBD.multiply(sqrtBD, new MathContext(99999, RoundingMode.HALF_UP))));
+            assertNumEquals(numOf(numBD), sqrt.multipliedBy(sqrt));
+        } else if (numOf(0).getClass().equals(PrecisionNum.class)) {
+            Properties props = new Properties();
+            try (InputStream is = getClass().getResourceAsStream("numTest.properties")) {
+                props.load(is);
+                assertNumEquals(props.getProperty("sqrtCorrect100000"), sqrt);
+                assertNumNotEquals(props.getProperty("sqrtCorrect99999"), sqrt);
+                assertNumEquals(Double.MAX_VALUE, sqrt);
+                assertNumNotEquals(numOf(Double.MAX_VALUE), sqrt);
+                BigDecimal sqrtBD = new BigDecimal(sqrt.toString());
+                assertNumEquals(numOf(numBD), numOf(sqrtBD.multiply(sqrtBD, new MathContext(99999, RoundingMode.HALF_UP))));
+                assertNumNotEquals(numOf(numBD), sqrt.multipliedBy(sqrt));
+            } catch (IOException ioe) {
+                ioe.printStackTrace();
```

Please let the test fail with a message if IOException has been thrown. I think you can call fail("no test properties found") to let a junit test fail